polyfill/lib/ecmascript.mjs

```diff
@@ -107,32 +107,20 @@ export const ES = ObjectAssign(ObjectAssign({}, ES2019), {
   },
   ToDateTime: (item) => {
     if (ES.IsDateTime(item)) return item;
-    const props = ES.ValidPropertyBag(item, [
-      'year',
-      'month',
-      'day',
-      'hour',
-      'minute',
-      'second',
-      'millisecond',
-      'microsecond',
-      'nanosecond'
-    ]);
-    if (props) {
-      const {
-        year = 0,
-        month = 1,
-        day = 1,
-        hour = 0,
-        minute = 0,
-        second = 0,
-        millisecond = 0,
-        microsecond = 0,
-        nanosecond = 0
-      } = props;
-      return new TemporalDateTime(year, month, day, hour, minute, second, millisecond, microsecond, nanosecond, 'reject');
-    }
-    const isoString = ES.ToString(item);
+    const {
+      year = 0,
+      month = 1,
+      day = 1,
+      hour = 0,
+      minute = 0,
+      second = 0,
+      millisecond = 0,
+      microsecond = 0,
+      nanosecond = 0,
+    } = item;
```

Comment:
Somewhat obscure, but the spec intentionally requires that the `Get` calls are done in lexicographical order. Then again, the current code doesn't do that either.

Reply:
I'll add another commit fixing this.